github/repos.go

```diff
@@ -1321,3 +1321,28 @@ func (s *RepositoriesService) Transfer(ctx context.Context, owner, repo string,
 
 	return r, resp, nil
 }
+
+// RepositoryDispatch represents a repository_dispatch event.
+type RepositoryDispatch struct {
+	EventType     string            `json:"event_type"`
+	ClientPayload map[string]string `json:"client_payload"`
+}
+
+// CreateRepoDispatch triggers a webhook event called repository_dispatch
```

Comment:
Please add a unit test for this new endpoint, following the pattern for unit testing other endpoints.